WalletWasabi.Tests/UnitTests/Transactions/TransactionProcessorTests.cs

```diff
@@ -157,7 +157,7 @@ public async Task ProcessResultAfterConfirmationCorrectAsync()
 		Assert.Empty(res4.NewlyConfirmedReceivedCoins);
 		Assert.Single(res4.NewlyConfirmedSpentCoins);
 		Assert.Empty(res4.NewlyReceivedCoins);
-		Assert.Empty(res4.NewlySpentCoins);
+		Assert.Single(res4.NewlySpentCoins);
```

Comment:
This is really interesting, and is maybe a problem on `master`: What happens here is that a coin was spent by TX3. Then, TX4 replaced TX3 and spent the same coin. 

The `TransactionProcessor` doesn't consider this coin as `NewlySpent`, because its field `SpenderTransaction` was `not null` (as TX3 spent it). However, it was `not null` only because we don't update this field in the `Undo` function, and this is probably a mistake.

This coin should arguably be `considered` NewlySpent as the TX that previously spent it doesn't exist anymore

Reply:
Fixed on `master`